plugins/unicodelib/unicodelib.cpp

```diff
@@ -719,6 +720,34 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+void translate(UnicodeString & source, UChar const * sear, unsigned searLen, UChar const * repl, unsigned replLen)
+{
+    UnicodeString search(sear, searLen);
+    UnicodeString replace(repl, replLen);
+    if (search.countChar32() != replace.countChar32() || source.isEmpty() || search.isEmpty() || replace.isEmpty())
+    {
+        return;
+    }
+    StringCharacterIterator it(source);
+    source.remove();
+    int32_t idx = it.setToStart();
+    while (idx != it.endIndex())
+    {
+        int32_t x = search.lastIndexOf(it.current32());
```

Comment:
I've edited the code so that it looks like this:

`    while (idx != CharacterIterator::DONE)
    {
        idx = it.next32PostInc();
        int32_t x = search.lastIndexOf(it.current32());
        if (x == -1)
        {
            toProcess.append(it.current32());
        }
        else
        {
            x = replace.moveIndex32(0, x);
            toProcess.append(replace.char32At(x));
        }
    }`

All of the test cases fail with this configuration with the addition of 2 \032 characters to the edited toProcess string.

`ctx->addWuAssertFailure(100000U,"Assert (uvw def \032\032! =  uvw def !) failed [translate(U' xyz abc ', U'xyzabc', U'uvwdef') + U'!' = U' uvw def !']","c:\\users\\skafda01\\hpcc-platform\\ecllibrary\\teststd\\uni\\TestTranslate.ecl",28U,29U,false);
				ctx->addWuAssertFailure(100000U,"Assert (uvw abc \032\032! =  uvw abc !) failed [translate(U' xyz abc ', U'zyx', U'wvu') + U'!' = U' uvw abc !']","c:\\users\\skafda01\\hpcc-platform\\ecllibrary\\teststd\\uni\\TestTranslate.ecl",29U,29U,false);`

I'm not sure why this is happening.